### ISSUE TITLE
1.18 fabric not launching

### ISSUE THREAD

**Issue by cootshk** [itzg/docker-minecraft-server#1167]
### Describe the problem

server unable to launch fabric 1.18

### Container definition

```
Paste run command or compose file here
```
sudo docker run -d -p 25565:25565 -p 25565:25565/udp -v /____:/data -e EULA=TRUE -e TYPE=fabric -e SERVRE_NAME=HypixelBestOnDotEighteen -e OPS=______ -e MAX_PLAYERS=50 -e FORCE_GAMEMODE=true -e SPAWN_PROTECTION=1 -e MOTD="Sodium on 1.18" -e ALLOW_FLIGHT=TRUE -e PLAYER_IDLE_TIMEOUT=5 -e OP_PERMISSION_LEVEL=4 -e MAX_MEMORY=8G --name=mc6 itzg/minecraft-server

### Container logs

```
Paste logs here
```

[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 3 1000 1000 4096 Dec  7 03:20 /data'
[init] Resolved version given LATEST into 1.18
[init] Resolving type given fabric
[init] Checking Fabric version information.
[init] server.properties already created, skipping
[init] Updating ops
[init] Checking for JSON files.
[init] Setting initial memory to 1G and max to 8G
[init] Starting the Minecraft server...
[03:21:13] [ERROR] [FabricLoader/]: Uncaught exception in thread "main"
java.lang.RuntimeException: An exception occurred when launching the server!
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:68)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:66)
Caused by: java.lang.RuntimeException: Error invoking MC server bundler: java.lang.UnsupportedClassVersionError: net/minecraft/bundler/Main has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 60.0
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.processBundlerJar(MinecraftGameProvider.java:318)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.locateGame(MinecraftGameProvider.java:203)
        at net.fabricmc.loader.impl.launch.knot.Knot.createGameProvider(Knot.java:172)
        at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:125)
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:71)
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
        ... 5 more
Caused by: java.lang.UnsupportedClassVersionError: net/minecraft/bundler/Main has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 60.0
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
        at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider$1.loadClass(MinecraftGameProvider.java:274)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Unknown Source)
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.processBundlerJar(MinecraftGameProvider.java:294)
        ... 10 more

**Comment by cootshk** [comment]
trying the same command, but making a vanilla server results in this:

[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 3 1000 1000 4096 Dec  7 03:20 /data'
[init] Resolved version given LATEST into 1.18
[init] Resolving type given VANILLA
[init] Downloading minecraft_server.1.18.jar ...
[init] server.properties already created, skipping
[init] Updating ops
[init] Checking for JSON files.
[init] Setting initial memory to 1G and max to 8G
[init] Starting the Minecraft server...
Error: LinkageError occurred while loading main class net.minecraft.bundler.Main
        java.lang.UnsupportedClassVersionError: net/minecraft/bundler/Main has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 60.0
2021-12-07T03:26:16.830Z        WARN    mc-server-runner        sub-process failed      {"exitCode": 1}
2021-12-07T03:26:16.831Z        INFO    mc-server-runner        Done

**Comment by itzg** [comment]
For both cases make sure to re-pull the latest image or use the `java17` image to be sure that Java 17 is being used.

**Comment by Minocula** [comment]
Hey Geoff, long time no see. I have the same / simular error here. My running container was working. I only start him when I need a fabric server. This was the old docker run command. 

docker run -d -p 25567:25565 -p 8125:8123 --name minecraft-java-fabric --net=minecraft_default -v minecraft_java_fabric:/data -e EULA=TRUE -e MEMORY=2G -e VERSION=1.17.1 -e TYPE=FABRIC itzg/minecraft-server:java16

I have re-pull java16 and java17

I always got the same Error:

[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxrwx 10 1000 1000 4096 Dec  7 09:16 /data'
[init] Resolved version given 1.17.1 into 1.17.1
[init] Resolving type given FABRIC
[init] Checking Fabric version information.
[init] Checking Fabric Loader version information.
[init] Downloading https://maven.fabricmc.net/net/fabricmc/fabric-installer/0.10.2/fabric-installer-0.10.2.jar ...
[init] Installing Fabric 1.17.1 using fabric-installer-0.10.2.jar with loader version 0.12.8
Loading Fabric Installer: 0.10.2
Installing Fabric Loader 0.12.8(1.17.1) on the server
wnloading required files
Downloading library net.fabricmc:tiny-mappings-parser:0.3.0+build.17
Downloading library net.fabricmc:sponge-mixin:0.10.7+mixin.0.8.4
Downloading library net.fabricmc:tiny-remapper:0.6.0
Downloading library net.fabricmc:access-widener:2.0.1
Downloading library org.ow2.asm:asm:9.2
Downloading library org.ow2.asm:asm-analysis:9.2
Downloading library org.ow2.asm:asm-commons:9.2
Downloading library org.ow2.asm:asm-tree:9.2
Downloading library org.ow2.asm:asm-util:9.2
Downloading library net.fabricmc:intermediary:1.17.1
Downloading library net.fabricmc:fabric-loader:0.12.8
Generating server launch JAR
Downloading Minecraft server
Done
Done, start server by running fabric-server-launch.jar
[init] server.properties already created, skipping
[init] Checking for JSON files.
[init] Setting initial memory to 2G and max to 2G
[init] Starting the Minecraft server...
[09:19:00] [INFO] [FabricLoader/GameProvider]: Loading for game Minecraft 1.17.1
[09:19:01] [ERROR] [FabricLoader/]: Uncaught exception in thread "main"
java.lang.RuntimeException: An exception occurred when launching the server!
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:68)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:66)
Caused by: java.lang.NoClassDefFoundError: org/apache/logging/log4j/LogManager
	at net.fabricmc.loader.impl.game.minecraft.Log4jLogHandler.getLogger(Log4jLogHandler.java:45)
	at net.fabricmc.loader.impl.game.minecraft.Log4jLogHandler.log(Log4jLogHandler.java:37)
	at net.fabricmc.loader.impl.util.log.BuiltinLogHandler.replay(BuiltinLogHandler.java:76)
	at net.fabricmc.loader.impl.util.log.Log.init(Log.java:34)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.setupLog4jLogHandler(MinecraftGameProvider.java:515)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.initialize(MinecraftGameProvider.java:498)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:136)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:71)
	at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23)
	... 5 more
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.LogManager
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:158)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)

	... 14 more
2021-12-07T09:19:01.149Z	WARN	mc-server-runner	sub-process failed	{"exitCode": 1}
2021-12-07T09:19:01.150Z	INFO	mc-server-runner	Done

Hope my answer can help.

**Comment by itzg** [comment]
@Minocula Yours seems to be failing for a different reason

```
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.LogManager
```

whereas the OP is

```
Caused by: java.lang.UnsupportedClassVersionError: net/minecraft/bundler/Main has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 60.0
```

**Comment by Ziondaman** [comment]
hi i head the same problem and found a way to fix it
first i started the server.jar
where it made the 1.18.1 sever files
then i started the 
fabric-server-launch.jar 
and it made the mods folder and extra
then i fix the eula to true
and started a bach flie that head the following
java -Xmx2G -jar fabric-server-launch.jar nogui
pause
and it started fullly

Stuff that i installed

- java 17
- fabric installer (i got both fabric-server-launch.jar  and  server.jar)

and i was running win 10
then movid it over to a lunix distro 

hope this helps
-ziondaman

**Comment by itzg** [comment]
The latest image now includes the fix from #1197. Please re-pull the image and give that a try.

**Comment by cootshk** [comment]
this fixes it, thanks!
